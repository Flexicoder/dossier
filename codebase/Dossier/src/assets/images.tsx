const images = {
    XD_emoticon: require('./img/XD_emoticon_2993603.png'),
    angel_emoticon: require('./img/angel_emoticon_2993658.png'),
    angry_emoticon_1: require('./img/angry_emoticon_1_2993656.png'),
    angry_emoticon_2: require('./img/angry_emoticon_2_2993655.png'),
    angry_emoticon_3: require('./img/angry_emoticon_3_2993654.png'),
    bored_emoticon: require('./img/bored_emoticon_2993652.png'),
    cool_emoticon: require('./img/cool_emoticon_2993651.png'),
    crazy_emoticon: require('./img/crazy_emoticon_2993650.png'),
    cry_emoticon_1: require('./img/cry_emoticon_1_2993649.png'),
    cry_emoticon_2: require('./img/cry_emoticon_2_2993648.png'),
    cry_emoticon_3: require('./img/cry_emoticon_3_2993647.png'),
    devil_emoticon: require('./img/devil_emoticon_2993646.png'),
    disgusted_emoticon_1: require('./img/disgusted_emoticon_1_2993645.png'),
    disgusted_emoticon_2: require('./img/disgusted_emoticon_2_2993644.png'),
    dislike_emoticon: require('./img/dislike_emoticon_2993643.png'),
    drool_emoticon: require('./img/drool_emoticon_2993642.png'),
    evil_laugh_emoticon: require('./img/evil_laugh_emoticon_2993638.png'),
    evil_stretch_tongue_emoticon: require('./img/evil_stretch_tongue_emoticon_2993641.png'),
    facepalm_emoticon: require('./img/facepalm_emoticon_2993640.png'),
    grumpy_emoticon: require('./img/grumpy_emoticon_2993639.png'),
    happy_poo_emoticon: require('./img/happy_poo_emoticon_2993637.png'),
    headache_emoticon: require('./img/headache_emoticon_2993635.png'),
    huffish_emoticon: require('./img/huffish_emoticon_2993634.png'),
    hypnotized_emoticon: require('./img/hypnotized_emoticon_2993633.png'),
    ill_emoticon: require('./img/ill_emoticon_2993632.png'),
    in_love_emoticon: require('./img/in_love_emoticon_2993598.png'),
    in_pain_emoticon: require('./img/in_pain_emoticon_2993631.png'),
    injured_emoticon: require('./img/injured_emoticon_2993588.png'),
    kiss_emoticon_1: require('./img/kiss_emoticon_1_2993630.png'),
    kiss_emoticon_2: require('./img/kiss_emoticon_2_2993602.png'),
    kiss_emoticon_3: require('./img/kiss_emoticon_3_2993589.png'),
    laugh_cry_emoticon: require('./img/laugh_cry_emoticon_2993601.png'),
    laugh_emoticon_1: require('./img/laugh_emoticon_1_2993628.png'),
    laugh_emoticon_2: require('./img/laugh_emoticon_2_2993629.png'),
    like_emoticon: require('./img/like_emoticon_2993584.png'),
    money_emoticon: require('./img/money_emoticon_2993591.png'),
    mustache_emoticon: require('./img/mustache_emoticon_2993626.png'),
    nerd_emoticon: require('./img/nerd_emoticon_2993590.png'),
    nervous_emoticon_1: require('./img/nervous_emoticon_1_2993582.png'),
    nervous_emoticon_2: require('./img/nervous_emoticon_2_2993625.png'),
    pirate_emoticon: require('./img/pirate_emoticon_2993624.png'),
    puke_emoticon: require('./img/puke_emoticon_2993587.png'),
    rolling_eyes_emoticon: require('./img/rolling_eyes_emoticon_2993585.png'),
    rotfl_emoticon_2: require('./img/rotfl_emoticon_2_2993600.png'),
    sad_emoticon_1: require('./img/sad_emoticon_1_2993623.png'),
    sad_emoticon_2: require('./img/sad_emoticon_2_2993622.png'),
    scared_emoticon: require('./img/scared_emoticon_2993620.png'),
    scared_to_death_emoticon: require('./img/scared_to_death_emoticon_2993595.png'),
    shocked_emoticon_1: require('./img/shocked_emoticon_1_2993621.png'),
    shocked_emoticon_2: require('./img/shocked_emoticon_2_2993594.png'),
    shy_emoticon: require('./img/shy_emoticon_2993619.png'),
    sick_emoticon: require('./img/sick_emoticon_2993618.png'),
    silent_emoticon: require('./img/silent_emoticon_2993586.png'),
    sleeping_emoticon: require('./img/sleeping_emoticon_2993592.png'),
    smile_emoticon_1: require('./img/smile_emoticon_1_2993617.png'),
    smile_emoticon_2: require('./img/smile_emoticon_2_2993616.png'),
    smile_emoticon_3: require('./img/smile_emoticon_3_2993615.png'),
    snoring_emoticon: require('./img/snoring_emoticon_2993614.png'),
    stop_emoticon: require('./img/stop_emoticon_2993593.png'),
    stressed_emoticon: require('./img/stressed_emoticon_2993597.png'),
    stretch_tongue_emoticon_1: require('./img/stretch_tongue_emoticon_1_2993613.png'),
    stretch_tongue_emoticon_2: require('./img/stretch_tongue_emoticon_2_2993596.png'),
    thinking_emoticon: require('./img/thinking_emoticon_2993612.png'),
    ugly_emoticon: require('./img/ugly_emoticon_2993610.png'),
    unamused_emoticon: require('./img/unamused_emoticon_2993611.png'),
    waveing_emoticon: require('./img/waveing_emoticon_2993579.png'),
    weird_emoticon_1: require('./img/weird_emoticon_1_2993609.png'),
    weird_emoticon_2: require('./img/weird_emoticon_2_2993607.png'),
    wink_emoticon_1: require('./img/wink_emoticon_1_2993608.png'),
    wink_emoticon_2: require('./img/wink_emoticon_2_2993606.png'),
    wistling_emoticon: require('./img/wistling_emoticon_2993605.png'),
    wondering_emoticon: require('./img/wondering_emoticon_2993604.png'),
    zipped_emoticon: require('./img/zipped_emoticon_2993599.png'),
}

export default images