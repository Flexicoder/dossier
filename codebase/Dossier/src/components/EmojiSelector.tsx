import React, { Component } from 'react';
import { Image, View } from 'react-native';
import  styles  from '../assets/styles'
import images from '../assets/images';

export interface EmojiProps {
    src: any;
    value: number;
}

export interface EmojiOptionProps {
    selectedValues: number[];
    multiSelection?: boolean | false;
    options: Array<EmojiProps>;
}


export default class EmojiSelector extends Component<EmojiOptionProps> {
    constructor(props: EmojiOptionProps) {
        super(props)
        this.state = { selectedValues: props.selectedValues,
                       multiSelection: props.multiSelection,
                       options: props.options }
    }

    render() {
        return (
            <View style={styles.wrapLine}>
                {this.props.options.map(emoji => {
                    return (<View style={styles.emotContainer}>
                                <Image source={emoji.src}  style={styles.emoticon} key={emoji.value} /> 
                                <Image source={images.empty} style={[styles.emoticonBorder, 
                                    this.props.selectedValues.includes(emoji.value) ? styles.emoticonSelected 
                                                                                    : styles.emoticonNotSelected]} />
                            </View>);
                })}
            </View>
        );
    }

}