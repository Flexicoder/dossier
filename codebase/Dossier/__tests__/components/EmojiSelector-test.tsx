
import 'react-native';
import React from 'react';
import EmojiSelector, * as emoji from '../../src/components/EmojiSelector';
import images from '../../src/assets/images'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe("EmojiSelector", () => {
    it('renders correctly', () => {
        const theseOptions: emoji.EmojiProps[] = [
            {
              src: images.crazy_emoticon,
              value: 0
            },
            {
              src: images.disgusted_emoticon_1,
              value: 1
            }
        ]
        const emojiSelector = renderer.create(<EmojiSelector options={theseOptions} />);
        expect(emojiSelector.toJSON()).toMatchSnapshot();
      });
})
