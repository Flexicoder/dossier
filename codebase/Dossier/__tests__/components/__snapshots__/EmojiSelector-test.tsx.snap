// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiSelector renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/img/crazy_emoticon_2993650.png",
        }
      }
      style={
        Object {
          "height": 30,
          "margin": 9,
          "resizeMode": "contain",
          "width": 30,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 38,
            "margin": 5,
            "position": "absolute",
            "resizeMode": "contain",
            "width": 38,
          },
          Object {
            "borderColor": "#d0d0d0",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/img/empty.png",
          }
        }
        style={
          Object {
            "opacity": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/img/disgusted_emoticon_1_2993645.png",
        }
      }
      style={
        Object {
          "height": 30,
          "margin": 9,
          "resizeMode": "contain",
          "width": 30,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 38,
            "margin": 5,
            "position": "absolute",
            "resizeMode": "contain",
            "width": 38,
          },
          Object {
            "borderColor": "#d0d0d0",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/img/empty.png",
          }
        }
        style={
          Object {
            "opacity": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
